src/lamp.py

```diff
@@ -11,18 +11,38 @@
 class Lamp:
     def __init__(self, name, base_color, shade_color):
         self.name = name
+        self.behaviours = []
         self.shade = LedStrip(shade_color, pixel_config['shade']['pin'], pixel_config['shade']['pixels'])
         self.base = LedStrip(base_color, pixel_config['base']['pin'], pixel_config['base']['pixels'])
+    
+    def add_behaviour(self,behaviour_class):
+        b = behaviour_class(self)
+        print(b)
+        asyncio.create_task(b.run())
+        
+        self.behaviours.append(b) 
+        print("Behaviour added: %s" % (b))
 
-        asyncio.run(self.wake())
- 
-    async def wake(self):
-        # This is currently not working; it waits until the first coroutine is done before moving on
+    # Reset the lamp to it's default colors
+    def reset(self):
+        self.shade.reset()
+        self.base.reset()
+
+    # Turn the lamp off
+    def off(self):
```

Comment:
Would turning off the lamp disable all behaviours, and turning it on would enable them?